### PR TITLE
port sample input smoke test

### PR DESCRIPTION
**Pull request by pmeier**
Builds on top of #7954. I'll rebase after that is merged.

cc @vfdev-5

### CODE REVIEW

**Comment by pmeier** [comment]
If we merge #8001 before this one, we can completely get rid of the old test.

**Comment by NicolasHug** [comment]
Sorry about the commit and revert, I was on the wrong branch

File: test/test_transforms_v2_refactored.py [hunk 1]
```diff
@@ -261,7 +263,133 @@ def _check_transform_v1_compatibility(transform, input, *, rtol, atol):
...
+def _make_transform_sample(transform, *, image_or_video, adapter):
+    device = image_or_video.device if isinstance(image_or_video, torch.Tensor) else "cpu"
+    size = F.get_size(image_or_video)
+    input = dict(
+        image_or_video=image_or_video,
+        image_tv_tensor=make_image(size, device=device),
+        video_tv_tensor=make_video(size, device=device),
+        image_pil=make_image_pil(size),
+        bounding_boxes_xyxy=make_bounding_boxes(
+            size, format=tv_tensors.BoundingBoxFormat.XYXY, num_objects=3, device=device
```
Comment (NicolasHug) [inline test/test_transforms_v2_refactored.py:275]: I know it comes from the original test, but we don't seem to have a good reason to set `num_objects` (previously `batch_dims)`?

Maybe we can avoid it (and avoid introducing the parameter in `make_bounding_boxes`)?

**Review by NicolasHug — APPROVED** [review]
Thanks a lot @pmeier

File: test/test_transforms_v2_refactored.py [hunk 2]
```diff
@@ -261,7 +263,133 @@ def _check_transform_v1_compatibility(transform, input, *, rtol, atol):
...
+        object=object(),
+        tensor=torch.empty(5),
+        array=np.empty(5),
+    )
+    if adapter is not None:
+        input = adapter(transform, input, device)
+    return input
+
+
+def _check_transform_sample_input_smoke(transform, input, *, adapter):
```
Comment (NicolasHug) [inline test/test_transforms_v2_refactored.py:345]: As discussed offline I can't think of an alternative name for this test, even though I feel like there must be a better one. Anyway, at least a small comment as below will help understanding the purpose (and the naming)

```suggestion
def _check_transform_sample_input_smoke(transform, input, *, adapter):
    # This is a bunch of input / output convention checks, using a big sample with different parts as input.
```

**Comment by pmeier** [comment]
Some benchmarks:

- `main`:

  ```shell
  $ pytest test/test_transforms_v2.py::TestSmoke::test_common test/test_transforms_v2_refactored.py
  [...]
  === 7928 passed, 34 skipped, 96 warnings in 19.48s ===
  ```
- PR:
  
  ```shell
  $ pytest test/test_transforms_v2_refactored.py
  ========= 6260 passed, 34 skipped in 16.01s ==========
  ```

Although we now test multiple configurations per transform, we ultimately end up (insignificantly on our scales) faster than before, because we removed some duplication between the legacy and current tests as well as doing some refactoring that was not possible in the legacy tests.

**Pull request merged**